pywbem_mock/_dmtf_schema.py

```diff
@@ -0,0 +1,392 @@
+  marked experimantel.  It is considered the stable release.
+
+* Experimental - Contains the classes in the 'Final' zip file plus other
+  classes that are considered experimental. Typically it is considered less
+  stable.
+
+:class:`DMTFSchema` will optionally load the experimental schema if the
+constructor parameter 'user-experimental' is 'True'.
+
+The DMTFSchema class also provides a method to create a MOF include file
+from the DMTF Schema downloaded that includes only a subset of the classes
+actually in the repository for compile into a MOF repository.
+"""
+import os
+from zipfile import ZipFile
+import shutil
+import six
+
+if six.PY2:
+    # pylint: disable=wrong-import-order
+    from urllib2 import urlopen
+else:
+    # pylint: disable=wrong-import-order
+    from urllib.request import urlopen
+
+
+__all__ = ['DMTFSchema']
+
+
+class DMTFSchema(object):
+    """
+    This class manages the download and install of these files into a form
+    that can be used by pywbem_mock to create a class and qualifier declaration
+    repository.
+
+    It also provides tools to build repositories with DMTF schema.
+    """
+    def __init__(self, schema_ver, schema_dir, use_experimental=False,
+                 verbose=False):
+        """
+        The constructor downloads the schema if the 'schema_dir' does not exist
+        and installs it into the directory defined by the 'schema_dir'
+        attribute.
+
+        If the schema is already installed it simply sets the property values so
+        that the schema can be used.
+
+        If the schema was previously downloaded but not expanded, it is
+        automatically expanded (i.e. the schema/mof subdirectory is created
+        and the mof files expanded into this directory.). This allows only
+        the single file 'schema_mof_filename' to be persisted and the expansion
+        to occur on usage which saves downloading from the DMTF repeatedly and
+        saves disk space.
+
+        Parameters:
+
+          schema_dir (:term:`string`):
+            Directory into which the DMTF schema is installed or will be
+            installed.
+
+          schema_ver  (tuple of 3 integers (x, y, z):
+            Represents the DMTF the schema version in the format x.yy.z where
+                x is the major version
+                y is the minor version
+                z is the revisions
+            This MUST represent a DMTF schema that is available from the
+            DMTF web site.
+
+          use_experimental (:class:`py:bool`):
+            If True the expermintal version of the defined schema is
+            installed.
+
+            If False (the default) the Final released version of the DMTF
+            is installed.
+
+          verbose (:class:`py:bool`):
+            If 'True' status messages are output to the console
+
+        Raises:
+            ValueError if the schema cannon be retrieved from the DMTF web site.
```

Comment:
If you use the syntax:

class: description

then Sphinx recognizes the exception class name and generates different output.

Occurs multiple times.

Reply:
Note that we have both the detailed definition :exc:`~py:exceptions.ValueError`: and the more general ValueError: in different parts of the documentaiton.

Fixed DONE